src/script/components/toggle/ReceiptModeToggle.tsx

```diff
@@ -28,20 +28,25 @@ import * as Icon from '../Icon';
 export interface ReceiptModeToggleProps {
   onReceiptModeChanged: (receiptMode: RECEIPT_MODE) => void;
   receiptMode: RECEIPT_MODE;
+  disabled?: boolean;
 }
 
-const ReceiptModeToggle = ({receiptMode, onReceiptModeChanged}: ReceiptModeToggleProps) => {
+const ReceiptModeToggle = ({receiptMode, onReceiptModeChanged, disabled = false}: ReceiptModeToggleProps) => {
   const updateValue = () => {
-    const newReceiptMode = receiptMode !== RECEIPT_MODE.ON ? RECEIPT_MODE.ON : RECEIPT_MODE.OFF;
-    onReceiptModeChanged(newReceiptMode);
+    if (!disabled) {
+      const newReceiptMode = receiptMode !== RECEIPT_MODE.ON ? RECEIPT_MODE.ON : RECEIPT_MODE.OFF;
+      onReceiptModeChanged(newReceiptMode);
+    }
   };
 
   const inputRef = React.useRef<HTMLInputElement>(null);
   const isChecked = receiptMode !== RECEIPT_MODE.OFF;
 
   return (
     <>
-      <div className="panel__action-item panel__action-item--toggle">
+      <div
+        className={`panel__action-item panel__action-item--toggle${disabled ? ' panel__action-item--disabled' : ''}`}
```

Comment:
suggestion: you can use classnames to make it more readable:

```suggestion
import cx from 'classnames';

        className={cx('panel__action-item', 'panel__action-item--toggle', {
         'panel__action-item--disabled': disabled
        })}
```